synapse/push/pusherpool.py

```diff
@@ -243,6 +258,14 @@ def _start_pushers(self):
         """
         pushers = yield self.store.get_all_pushers()
 
+        pushers = [
+            pusherdict
+            for pusherdict in pushers
+            if self._pusher_shard_config.should_handle(
+                self._instance_name, pusherdict["user_name"]
+            )
+        ]
```

Comment:
It probably doesn't matter, but we end up doing this work twice unfortunately since `_start_pusher` has the same guard in front of it.